### PR TITLE
Improve radio button demos

### LINKED ISSUE #119: Improve vaadin-radio-button demos

### PR DESCRIPTION
**Pull request by shadikhani**
Fix #119

### CODE REVIEW

File: demo/radio-button-basic-demos.html [hunk 1]
```diff
@@ -5,22 +5,93 @@
         display: block;
       }
     </style>
+    <h3>Basic usage</h3>
+    <vaadin-demo-snippet id="basic-usage">
+      <template preserve-content>
+        <vaadin-radio-group label="Label" theme="vertical">
```
Comment (web-padawan) [inline demo/radio-button-basic-demos.html]: Why do we yse `theme="vertical"` from almost all the examples now?
I think it should be only used in "Theme Variants".

File: demo/radio-button-basic-demos.html [hunk 2]
```diff
@@ -5,22 +5,93 @@
...
+          <vaadin-radio-button checked>Option one</vaadin-radio-button>
+          <vaadin-radio-button>Option two</vaadin-radio-button>
+          <vaadin-radio-button>Option three</vaadin-radio-button>
+        </vaadin-radio-group>
+      </template>
+    </vaadin-demo-snippet>
+
+    <h3>Disabled and disabled item</h3>
```
Comment (web-padawan) [inline demo/radio-button-basic-demos.html]: I would recommend to shorten this to use "Disabled state". 
Users should be able to recognise the difference.

File: demo/radio-button-basic-demos.html [hunk 3]
```diff
@@ -5,22 +5,93 @@
...
+    <vaadin-demo-snippet id="disabled-and-disabled-item">
+      <template preserve-content>
+        <vaadin-radio-group label="Disabled" disabled theme="vertical"
+        >
+        <vaadin-radio-button checked>Option one</vaadin-radio-button>
+        <vaadin-radio-button>Option two</vaadin-radio-button>
+        <vaadin-radio-button>Option three</vaadin-radio-button>
+      </vaadin-radio-group>
 
+      <span style="margin-right: 200px;"></span>
```
Comment (web-padawan) [inline demo/radio-button-basic-demos.html]: You can remove this also when removing `theme="vertical"`

File: demo/radio-button-basic-demos.html [hunk 4]
```diff
@@ -5,22 +5,93 @@
...
+          <vaadin-radio-button>Option three</vaadin-radio-button>
+        </vaadin-radio-group>
+      </template>
+    </vaadin-demo-snippet>
+
+    <h3>Disabled and disabled item</h3>
+    <vaadin-demo-snippet id="disabled-and-disabled-item">
+      <template preserve-content>
+        <vaadin-radio-group label="Disabled" disabled theme="vertical"
+        >
```
Comment (web-padawan) [inline demo/radio-button-basic-demos.html]: Please move the closing bracket to the line above.

File: demo/radio-button-basic-demos.html [hunk 5]
```diff
@@ -5,22 +5,93 @@
...
-    <vaadin-demo-snippet id="radio-button-basic-demos-default-radio">
+      <vaadin-radio-group label="Team" theme="vertical">
+        <vaadin-radio-button>Option one</vaadin-radio-button>
+        <vaadin-radio-button>Option two</vaadin-radio-button>
+        <vaadin-radio-button disabled>Option three</vaadin-radio-button>
+      </vaadin-radio-group>
+    </template>
+  </vaadin-demo-snippet>
+
+    <h3>Read-only</h3>
```
Comment (web-padawan) [inline demo/radio-button-basic-demos.html]: This demo snippet and all the snippets below should be properly indented.

File: demo/radio-button-basic-demos.html [hunk 6]
```diff
@@ -5,22 +5,93 @@
...
+    <vaadin-demo-snippet id="read-only" when-defined="vaadin-radio-button">
```
Comment (web-padawan) [inline demo/radio-button-basic-demos.html]: You only need `when-defined="vaadin-radio-button"` if the demo snippet contains a script. 
Otherwise, you can remove it.

File: demo/radio-button-presentation-demos.html [hunk 7]
```diff
@@ -0,0 +1,68 @@
+<dom-module id="radio-button-presentation-demos">
+  <template>
+    <style include="vaadin-component-demo-shared-styles">
+        :host {
+          display: block;
+        }
```
Comment (web-padawan) [inline demo/radio-button-presentation-demos.html]: The CSS is not properly indented. Please use 2 spaces.

File: demo/radio-button-styling-demos.html [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        }
+      </style>
+    <h3>Customizing font</h3>
+    <vaadin-demo-snippet id="customizing-font" when-defined="vaadin-radio-button">
+      <template preserve-content>
+        <style>
+              vaadin-radio-button {
+                font-weight: bold;
+                font-family: sans-serif;
+              }
```
Comment (web-padawan) [inline demo/radio-button-styling-demos.html]: The CSS is not properly indented. Please use 2 spaces.

File: demo/radio-button-basic-demos.html (hunk 1 above)
Comment (shadikhani) [reply demo/radio-button-basic-demos.html]: That is what we discussed with Jouni. He said that it's better to have the demos in a vertical position.

File: demo/demos.json [hunk 9]
```diff
@@ -1,48 +1,55 @@
...
-  {
-    "name": "Radio Button",
-    "url": "radio-button-basic-demos",
-    "src": "radio-button-basic-demos.html",
-    "meta": {
-      "title": "vaadin-radio-button Examples",
-      "description": "",
-      "image": ""
+    {
+      "name": "Basic Usage",
```
[the remark this answers is not in the document]
Comment (shadikhani) [reply demo/demos.json]: I have already changed the "Basic" demos to "Basic usage". You can also see the vaadin-select and vaadin-combo-box demos. If you think I should change them all in basic I should talk with Jouni too. Should I talk to him about this change?

[the remark this answers is not in the document]
Comment (web-padawan) [reply demo/demos.json]: If that was discussed with Jouni, then it's good to keep as it is.

File: demo/radio-button-basic-demos.html (hunk 1 above)
Comment (web-padawan) [reply demo/radio-button-basic-demos.html]: If that was discussed with Jouni, then let's keep it.

File: demo/radio-button-basic-demos.html (hunk 3 above)
Comment (shadikhani) [reply demo/radio-button-basic-demos.html]: It's related to my previous comments. I already discussed it with Jouni and it is fine.

File: demo/radio-button-basic-demos.html (hunk 2 above)
Comment (shadikhani) [reply demo/radio-button-basic-demos.html]: Done

File: demo/radio-button-basic-demos.html (hunk 4 above)
Comment (shadikhani) [reply demo/radio-button-basic-demos.html]: Done.

File: demo/radio-button-basic-demos.html (hunk 5 above)
Comment (shadikhani) [reply demo/radio-button-basic-demos.html]: Done.

File: demo/radio-button-basic-demos.html (hunk 6 above)
Comment (shadikhani) [reply demo/radio-button-basic-demos.html]: Done.

File: demo/radio-button-presentation-demos.html (hunk 7 above)
Comment (shadikhani) [reply demo/radio-button-presentation-demos.html]: Done.

File: demo/radio-button-styling-demos.html (hunk 8 above)
Comment (shadikhani) [reply demo/radio-button-styling-demos.html]: Done.

File: demo/index.html [hunk 10]
```diff
@@ -1,29 +1,39 @@
...
-  <link rel="import" href="../../vaadin-demo-helpers/vaadin-demo-snippet.html">
-  <script src="../../vaadin-demo-helpers/vaadin-demo-ready-event-emitter.js"></script>
+    <link rel="import" href="../../polymer/polymer.html">
+    <link rel="import" href="../../vaadin-demo-helpers/vaadin-component-demo.html">
+    <link rel="import" href="../../vaadin-demo-helpers/vaadin-demo-snippet.html">
+    <script src="../../vaadin-demo-helpers/vaadin-demo-ready-event-emitter.js"></script>
+    <script src="radio-button-demo.js"></script>
 
-  <script src="radio-button-demo.js"></script>
+    <link rel="import" href="../../iron-form/iron-form.html">
```
Comment (web-padawan) [inline demo/index.html]: You can remove the `iron-form` import as you no longer use it in demo.

File: demo/index.html [hunk 11]
```diff
@@ -1,29 +1,39 @@
...
 <html>
-<head>
-  <meta charset="utf-8">
-  <meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
-  <title>vaadin-radio-button Examples</title>
-  <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1"
+      />
```
Comment (web-padawan) [inline demo/index.html]: Please move this closing bracket to previous line.

File: demo/index.html (hunk 10 above)
Comment (shadikhani) [reply demo/index.html]: Done.

File: demo/index.html (hunk 11 above)
Comment (shadikhani) [reply demo/index.html]: Done.

File: demo/index.html [hunk 12]
```diff
@@ -1,29 +1,37 @@
...
-
-<body>
-  <vaadin-component-demo config-src="demos.json"></vaadin-component-demo>
-</body>
+  <body>
+    <vaadin-component-demo config-src="demos.json"></vaadin-component-demo>
+  </body>
 </html>
+
+<dom-module id="custom-radio-button-icon-color" theme-for="vaadin-radio-button">
```
Comment (web-padawan) [inline demo/index.html]: Can you please move `dom-module` tag inside `body`? 

It will be rendered by browser there anyways. Nothing should be outside `<html>`

Comment (shadikhani) [reply demo/index.html]: Done.
